File: client/src/domain/Game.ts
import Board from "./Board";

export default class Game {
  readonly id: string;
  readonly state: string;
  readonly board: Board;
  readonly hitsRemaining: number;
  readonly shipsDestroyed: number;

  constructor(
    id: string,
    state: string,
    board: Board,
    hitsRemaining: number,
    shipsDestroyed: number
  ) {
    this.id = id;
    this.state = state;
    this.board = board;
    this.hitsRemaining = hitsRemaining;
    this.shipsDestroyed = shipsDestroyed;
  }
}

File: server/battleships/domain/src/main/java/eu/optas/domain/Game.java
package eu.optas.domain;

import eu.optas.utils.GameState;

public class Game {
    private final String id;
    private final GameState state;
    private final Board board;
    private final int hitsRemaining;
    private final int shipsDestroyed;

    public Game(String id, GameState state, Board board, int hitsRemaining, int shipsDestroyed) {
        this.id = id;
        this.state = state;
        this.board = board;
        this.hitsRemaining = hitsRemaining;
        this.shipsDestroyed = shipsDestroyed;
    }

    public String getId() {
        return id;
    }

    public GameState getState() {
        return state;
    }

    public Board getBoard() {
        return board;
    }

    public int getHitsRemaining() {
        return hitsRemaining;
    }

    public int getShipsDestroyed() {
        return shipsDestroyed;
    }
}
